src/Console/ConfigurationResolver.php

```diff
@@ -358,23 +359,32 @@ public function getPath()
             if (1 === count($this->options['path']) && '-' === $this->options['path'][0]) {
                 $this->path = $this->options['path'];
             } else {
-                $this->path = array_map(
-                    function ($path) use ($cwd, $filesystem) {
-                        $absolutePath = $filesystem->isAbsolutePath($path)
-                            ? $path
-                            : $cwd.DIRECTORY_SEPARATOR.$path;
-
-                        if (!file_exists($absolutePath)) {
-                            throw new InvalidConfigurationException(sprintf(
-                                'The path "%s" is not readable.',
-                                $path
-                            ));
-                        }
+                $this->path = array();
+                foreach ($this->options['path'] as $path) {
+                    if ('' === $path) {
+                        // TODO  trigger deprecate/warning?
+                    }
 
-                        return $absolutePath;
-                    },
-                    $this->options['path']
-                );
+                    $absolutePath = $filesystem->isAbsolutePath($path)
+                        ? $path
+                        : $cwd.DIRECTORY_SEPARATOR.$path;
+
+                    $path = @realpath($absolutePath);
+                    if (false === $path) {
+                        throw new InvalidConfigurationException(sprintf(
+                            'The path "%s" is not readable.',
+                            $absolutePath
+                        ));
+                    }
+
+                    if (isset($this->path[$path])) {
+                        continue; // TODO trigger warning/deprecate?
```

Comment:
I would not break the flow if someone passed same file twice (now or later, so for sure not a deprecation).
It may be nice to show info on console (like we do with xdebug enabled), but nothing crucial, if we would not show anything and just fix file once - good for me as well.